src/Auth/BaseAuthenticate.php

```diff
@@ -110,15 +110,22 @@ protected function _findUser($username, $password = null)
             return false;
         }
 
+        $passwordField = $this->_config['fields']['password'];
         if ($password !== null) {
             $hasher = $this->passwordHasher();
-            $hashedPassword = $result->get($this->_config['fields']['password']);
+            $hashedPassword = $result->get($passwordField);
             if (!$hasher->check($password, $hashedPassword)) {
                 return false;
             }
 
             $this->_needsPasswordRehash = $hasher->needsRehash($hashedPassword);
-            $result->unsetProperty($this->_config['fields']['password']);
+            $result->unsetProperty($passwordField);
+        }
+        if ($password === null && in_array($passwordField, $result->getHidden())) {
+            $hidden = $result->getHidden();
```

Comment:
Couldn't this be moved above the if clause? This would result in ``getHidden()`` being called only once.